nuget-tools.cake

```diff
@@ -0,0 +1,224 @@
+{
+    var destinationPath = $"{tempContentPath}/lib/{frameworkName.GetShortFolderName()}/*.dll";
+    var files = GetFiles(destinationPath);
+    if (files.Any()) 
+    {
+        return files;
+    }
+    Warning($"Haven't found anything under {tempContentPath} - make sure it's expected.");
+    return null;
+}
+
+void MoveDllsByArchitecture(string tempContentPath, string destination)
+{
+    var runtimesPath = tempContentPath + "/runtimes";
+    if (!DirectoryExists(runtimesPath)) 
+    {
+        return;
+    }
+    Information($"Move dlls by architecture from {tempContentPath} to {destination}");
+    foreach (var runtime in GetDirectories($"{runtimesPath}/win10-*")) {
+        var arch = runtime.GetDirectoryName().ToString().Replace("win10-", "").ToUpper();
+        var nativeFiles = GetFiles(runtime + "/**/*.dll");
+        var targetArchPath = destination + "/" + arch;
+        EnsureDirectoryExists(targetArchPath);
+        foreach (var nativeFile in nativeFiles) 
+        {
+            if (!FileExists(targetArchPath + "/" + nativeFile.GetFilename())) 
+            {
+                MoveFileToDirectory(nativeFile, targetArchPath);
+                Information($"Moved native binary file {nativeFile} to {targetArchPath}");
+            } 
+            else 
+            {
+                Information("Native binary file already exists");
+            }
+        }
+    } 
+}
+
+void MoveDlls(NugetDependency package, string tempContentPath, string destination)
+{
+    Information($"Move assemblies for {package.Name} from {tempContentPath} to {destination}.");
+    var dllFiles = ResolveDllFiles(tempContentPath, package.Framework);
+    if (dllFiles == null)
+    {
+        return;
+    }
+    foreach (var matchingFile in dllFiles) 
+    {
+        var targetPath = $"{destination}/{matchingFile.GetFilename()}";
+        if (FileExists(targetPath)) 
+        {
+            DeleteFile(targetPath);
+        }
+        MoveFile(matchingFile.FullPath, targetPath);
+        Information($"Moving {matchingFile.FullPath} to {targetPath}");
+    } 
+}
+
+void ProcessPackageDlls(NugetDependency package, string tempContentPath, string destination)
+{
+    Information($"Process {package.Name} dlls. Temp content path is {tempContentPath}; Destination is {destination}");
+    MoveDllsByArchitecture(tempContentPath, destination);
+    MoveDlls(package, tempContentPath, destination);
+}
+
+void ProcessAppCenterDlls(string tempContentPath, string destination) 
+{
+    MoveDllsByArchitecture(tempContentPath, destination);
+    var contentPathSuffix = "lib/uap10.0/";
+    var filesMask = tempContentPath + contentPathSuffix + '*';
+    Information($"Moving SDK package, move from {filesMask} to {destination}");
+    var files = GetFiles(filesMask);
+    CleanDirectories(destination);
+    MoveFiles(files, destination);
+}
+
+private void ExtractPackage(string fileName, string targetPath)
```

Comment:
BTW, why not just unzip it? it's temporary data anyway

Reply:
You're right.